pkg/nfd-master/nfd-master.go

```diff
@@ -381,6 +390,10 @@ func (m *nfdMaster) nfdAPIUpdateHandler() {
 			} else {
 				for nodeName := range updateNodes {
 					m.nodeUpdaterPool.queue.Add(nodeName)
+					if err := m.nfdAPIUpdateOneNode(nodeName); err != nil {
+						klog.ErrorS(err, "failed to update node", "node", nodeName)
+						updateNodes[nodeName] = struct{}{}
+					}
```

Comment:
Re-iterating https://github.com/kubernetes-sigs/node-feature-discovery/pull/1242#pullrequestreview-1471971539.

We definitely don't want to do this. nodeUpdaterPool does the node updates for us so we need to hook there